src/test/java/teammates/test/driver/HtmlHelper.java

```diff
@@ -112,146 +89,147 @@ private static Node getNodeFromString(String string) throws SAXException, IOExce
+     * <li><code>div</code> with id <code>student-motd-container</code>, which is used
+     *     as a container to place the MOTD</li>
+     * <li>Embedded <code>script</code> with variable <code>motdUrl</code>, which is used
+     *     to indicate where to find the HTML file for MOTD</li>
+     * </ul>
+     * TODO check if wildcarding this in place of ignoring is better
+     */
+    private static boolean isMotdComponent(Node currentNode) {
+        String currentNodeName = currentNode.getNodeName().toLowerCase();
+        if (currentNodeName.equalsIgnoreCase("script") || currentNodeName.equalsIgnoreCase("div")) {
             NamedNodeMap attributes = currentNode.getAttributes();
-            
-            if (attributes == null) { 
-                return false;
-            }
-                
             for (int i = 0; i < attributes.getLength(); i++) {
                 Node attribute = attributes.item(i);
-                
-                // Motd container
-                if (attribute.getNodeName().equalsIgnoreCase("id")
-                      && attribute.getNodeValue().contains("student-motd-container")) {
+                if (currentNodeName.equals("script")
+                        && attribute.getNodeName().equalsIgnoreCase("src")
+                        && attribute.getNodeValue().contains("studentMotd.js")) {
+                    return true;
+                } else if (currentNodeName.equals("div")
+                               && attribute.getNodeName().equalsIgnoreCase("id")
+                               && attribute.getNodeValue().contains("student-motd-container")) {
                     return true;
                 }
             }
+            
+            return currentNodeName.equals("script")
+                       && currentNode.getTextContent().contains("motdUrl");
         }
         
         return false;
     }
 
-    private static List<Node> getAttributesAsNodeList(NamedNodeMap actualAttributeList) {
-        List<Node> nodesList= new ArrayList<Node>();
-        for (int i = 0; i < actualAttributeList.getLength(); i++){
-            nodesList.add(actualAttributeList.item(i));
+    private static String getNodeOpeningTag(Node currentNode) {
+        StringBuilder openingTag = new StringBuilder();
+        // add the start of opening tag
+        openingTag.append("<" + currentNode.getNodeName().toLowerCase());
+        
+        // add the attributes of the tag (getAttributes() returns the attributes sorted alphabetically)
+        NamedNodeMap attributes = currentNode.getAttributes();
+        if (attributes != null) { // TODO check if this null check is still necessary
+            for (int i = 0; i < attributes.getLength(); i++) {
+                Node attribute = attributes.item(i);
+                openingTag.append(" " + attribute.getNodeName().toLowerCase() + "="
+                                      + "\"" + attribute.getNodeValue().replace("\"", "&quot;") + "\"");
```

Comment:
This is not for sanitization, but for `DOMParser` to be able to parse properly. Parsing something like `value="abc"def"` could be disastrous, so `value="abc&quot;def"`. Replacing with `'` (as before) doesn't make sense.